src/lib/defaultCommands.js

```diff
@@ -574,9 +622,17 @@ define(function (require) {
 						if (!editor.getRangeHelper().selectedHtml() || text) {
 							text = text || url;
 
-							editor.wysiwygEditorInsertHtml(
-								'<a href="' + url + '">' + text + '</a>'
-							);
+							// Replace existing Link, instead of stacking.
+							if ($anchor.length) {
+								if (text) {
+									$anchor.attr('href', text);
+								}
+								$anchor.attr('href', url);
```

Comment:
Could replace this with just `$anchor.attr('href', text || url);`. Maybe not quite as nice but it is a bit smaller.